### DISCUSSION TITLE
`loss_b_2` in positional loss has no `grad_fn`.

### DISCUSSION THREAD

**Discussion by mhjc01** [vijaydwivedi75/gnn-lspe#3] [Q&A]
https://github.com/vijaydwivedi75/gnn-lspe/blob/6b38443677befd4726cc0e5c20e9c88073fbc2ed/nets/ZINC_graph_regression/gatedgcn_net.py#L156

'loss_b_2' has no attributes of `grad_fn` and `requires_grad` as `False`, then, it give no effect in the gradient of positional embedding `p` when gradients of the loss propagate backward.

It is because the `P` and `PTP_In` is a scipy sparse matrix instead of a pytorch tensor.

Was that intended?
